### DISCUSSION TITLE
Is there a way to put cli_usage to custom header?

### DISCUSSION THREAD

**Discussion by niksfirefly** [bashly-framework/bashly#264] [Q&A]
Is there a way to put cli_usage to custom header?
I want text help message beneath shebang

regards

**Comment by DannyBen** [comment]
I am moving this to discussions.

**Reply by DannyBen** [reply]
Not sure I understand what you are trying to do.
If you want to replace the header of the generated script, you can do so - see [here](https://bashly.dannyb.co/usage/writing-your-scripts/#custom-header). There is an example for it [here](https://github.com/DannyBen/bashly/tree/master/examples/custom-script-header#readme).

If you want to show / run something before running any command, you can use the [initialize function](https://bashly.dannyb.co/usage/writing-your-scripts/#initialization-function).

If it is none of the above, I need more details with an example.

**Reply by niksfirefly** [reply]
Ok
I want header - of cli sample app - to look like this
```
#!/usr/bin/env bash
# cli - sample app

# Usage:
#   cli [command]
#   cli [command] --help | -h
#   cli --version | -v

# Commands:
#   help          Show help about a command
#   completions   Generate bash completions

# Options:
#   --help, -h
#     Show this help

#   --version, -v
#     Show version number
```
after each bashly generate

**Reply by DannyBen** [reply]
I see. There is no built in functionality for this. 
You can easily achieve it though by creating the `src/header.sh` prior to generating the final script.

For example, you can create a `generate.sh` script in your bashly root to achieve this task:

```bash
#!/usr/bin/env bash

bashly generate
printf "#!/usr/bin/env bash\n\n" > src/header.sh
./cli --help | sed -e "s/^/# /;" | sed -e "s/^# $/#/;" >> src/header.sh
bashly generate
```

**Reply by niksfirefly** [reply]
Nice, thanks
I got similar script called deploy.sh to do many due deploy script to production
regards
